src/runtime_src/core/edge/drm/zocl/zocl_rpu_channel.c

```diff
@@ -0,0 +1,159 @@
+/* SPDX-License-Identifier: GPL-2.0 OR Apache-2.0 */
+/*
+ * Copyright (C) 2021 Xilinx, Inc. All rights reserved.
+ *
+ * Author(s):
+ *        Lizhi Hou <lizhih@xilinx.com>
+ *
+ * This file is dual-licensed; you may select either the GNU General Public
+ * License version 2 or Apache License, Version 2.0.
+ */
+
+#include <linux/mod_devicetable.h>
+#include <linux/platform_device.h>
+#include <linux/of_address.h>
+#include "zocl_drv.h"
+
+#define ZRPU_CHANNEL_NAME "zocl_rpu_channel"
+
+#define ZCHAN2PDEV(chan)		((chan)->zrc_pdev)
+#define ZCHAN2DEV(chan)			(&ZCHAN2PDEV(chan)->dev)
+#define zchan_err(chan, fmt, args...)	zocl_err(ZCHAN2DEV(chan), fmt"\n", ##args)
+#define zchan_info(chan, fmt, args...)	zocl_info(ZCHAN2DEV(chan), fmt"\n", ##args)
+#define zchan_dbg(chan, fmt, args...)	zocl_dbg(ZCHAN2DEV(chan), fmt"\n", ##args)
+
+/* reserve 4k shared memory for RPU outband communication */
+#define ZRPU_CHANNEL_READY		0
```

Comment:
so it is like
struct zocl_rpu_status {
    u32 zocl_rpu_ready;
    u32 zocl_rpu_xgq_buf_size;
}

It's ok that I hardcode this in RPU also, but we'd better have this shared in a header file somewhere someday later.